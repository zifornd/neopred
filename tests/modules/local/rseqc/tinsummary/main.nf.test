nextflow_process {

    name "Test Process RSEQC_TINSUMMARY"
    script "modules/local/rseqc/tinsummary/main.nf"
    process "RSEQC_TINSUMMARY"

    test("rseqc tin summary for bam files") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of([
			   [id: 'test'],
			   file("$projectDir/tests/results/test.paired_end.sorted.summary.txt")
])
                """
            }
        }

        then {
            assert process.success
            assert process.out.summary_txt
	    assert process.out.versions
            assert snapshot(process.out).match()
        }

    }

}
