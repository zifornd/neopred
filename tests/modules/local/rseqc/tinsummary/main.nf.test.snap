{
    "rseqc tin summary for bam files": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tin_score_summary.txt:md5,4fd9c0d017704aba0be475e683807ce9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,694f88bca6264c9fe6301398e68cc636"
                ],
                "summary_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tin_score_summary.txt:md5,4fd9c0d017704aba0be475e683807ce9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,694f88bca6264c9fe6301398e68cc636"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-12-11T06:13:40.661676"
    }
}